### PR TITLE
CAppParamParserLinux: add --audio-backend=<backend> switch 

### PR DESCRIPTION
**Pull request by lrusak**
This changes the way we set the audio backend. This moves from using a (non-obvious) environment variable to a command line switch (`--audio-backend=`).

```
$ ./kodi.bin --help
Usage: kodi [OPTION]... [FILE]...

Arguments:
  -fs                   Runs Kodi in full screen
  --standalone          Kodi runs in a stand alone environment without a window
                        manager and supporting applications. For example, that
                        enables network settings.
  -p or --portable      Kodi will look for configurations in install folder instead of ~/.kodi
  --debug               Enable debug logging
  --version             Print version information
  --test                Enable test mode. [FILE] required.
  --settings=<filename> Loads specified file after advancedsettings.xml replacing any settings specified
                        specified file must exist in special://xbmc/system/

Linux Specific Arguments:
  --windowing=<system>  Select which windowing method to use.
                          Available window systems are: wayland, gbm, x11
  --logging=<target>    Select which log target to use (log file will always be used in conjunction).
                          Available log targets are: console
  --audio-backend=<backend> Select which audio backend to use.
                          Available audio backends are: pipewire
```

## What is the effect on users?

This allows users to select the audio backend with a command line switch when starting kodi. This also allow seeing the enabled audio backends that Kodi was built with.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -255,12 +255,33 @@ endif()
...
+endif()
+
+if(SNDIO_FOUND)
+  list(APPEND AUDIO_BACKENDS_LIST "sndio")
+endif()
+
+if(ALSA_FOUND AND PULSEAUDIO_FOUND)
+  list(APPEND AUDIO_BACKENDS_LIST "alsa+pulseaudio")
+endif()
+
```
Comment (fuzzard) [inline CMakeLists.txt:277]: is it better to append to this list in the respective find module? everything except the last one is pretty straight forward. 

For scope, you'll just need to init the list in the top CMakeLists.txt for ease

**Comment by wsnipex** [comment]
removing the env vars is going to break existing setups. Would be better to log a deprecation notice and only remove them in v22

**Comment by fritsch** [comment]
I agree with @wsnipex. Let's deprecate it first and then remove it in next major. While I defintely admit that no one might basically read the output, still it's good practice.

Rest of the code is fine and the right thing to have it proper.

File: xbmc/platform/linux/AppParamParserLinux.cpp [hunk 2]
```diff
@@ -82,12 +90,29 @@ void CAppParamParserLinux::ParseArg(const std::string& arg)
...
+      GetAppParams()->SetAudioBackend(*it);
+    }
+    else
+    {
+      std::cout << StringUtils::Format(audioBackendsText, arg.substr(15),
+                                       StringUtils::Join(availableAudioBackends, ", "));
+
+      exit(0);
+    }
+  }
```
Comment (neo1973) [inline xbmc/platform/linux/AppParamParserLinux.cpp:108]: This also fixes the `arg.substr(15)`:
```suggestion
  else if (arg.substr(0, 16) == "--audio-backend=")
  {
    const auto argValue = arg.substr(16);
    if (std::any_of(availableAudioBackends.cbegin(), availableAudioBackends.cend(),
                    [&](const auto& v) { return v == argValue; }))
    {
      GetAppParams()->SetAudioBackend(argValue);
    }
    else
    {
      std::cout << StringUtils::Format(audioBackendsText, argValue,
                                       StringUtils::Join(availableAudioBackends, ", "));

      exit(0);
    }
  }
```

Comment (lrusak) [reply xbmc/platform/linux/AppParamParserLinux.cpp]: partially updated.

**Comment by lrusak** [comment]
> removing the env vars is going to break existing setups. Would be better to log a deprecation notice and only remove them in v22

> I agree with @wsnipex. Let's deprecate it first and then remove it in next major. While I defintely admit that no one might basically read the output, still it's good practice.
> 

I don't really like the thought of having both env vars and cmdline switches. Setups either break now or in v22, what's the point in pushing off such a simple change? The env var isn't documented AFAIK so moving to the cmdline switch seems like a good move.

**Comment by fritsch** [comment]
It is used in libreelec and osmc iirc. Also the wiki documents it. Though I don't care too much if release documentation has it, but good practice it is never the less.

**Comment by wsnipex** [comment]
A simple backwards compatible fix for this is the [kodi.sh wrapper.](https://github.com/xbmc/xbmc/blob/master/tools/Linux/kodi.sh.in) Just catch the env vars there an add the proper command line switch.

**Review by wsnipex — APPROVED** [review]

**Comment by lrusak** [comment]
I'd like to merge this but I'm just wondering if anyone has anything better that "Audio Backend"?

Maybe:
 - Audio Driver
 - Audio Platform
 - Audio System
 - Sound Architecture

Suggestions welcome.

**Comment by DarkSphere00** [comment]
> I'd like to merge this but I'm just wondering if anyone has anything better that "Audio Backend"?
> 
> Maybe:
> 
>     * Audio Driver
> 
>     * Audio Platform
> 
>     * Audio System
> 
>     * Sound Architecture
> 
> 
> Suggestions welcome.

It's probably too late now, but what about audio/sound server? I think it's probably the most technically correct definition

**Comment by fritsch** [comment]
It would be correct for pulseaudio and pipewire, but not for OSS or ALSA.

**Pull request merged**